src/realm/list.hpp

```diff
@@ -928,10 +928,11 @@ inline LnkLst Obj::get_linklist(StringData col_name) const
 template <class T>
 void Lst<T>::clear()
 {
-    if (size() > 0) {
-        if (Replication* repl = Base::get_replication()) {
-            repl->list_clear(*this);
-        }
+    auto sz = size();
+    if (Replication* repl = Base::get_replication()) {
```

Comment:
This is a breaking change that we've held off for years because we weren't sure if folks were depending on the pre-existing behavior. How did this change fit into the bug this PR was fixing?

Reply:
I thought we specifically discussed doing this and concluded that we shouldn't make this change in a meeting a few weeks ago.

Reply:
it is going to be reverted for now